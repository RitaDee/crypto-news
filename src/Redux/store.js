import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './coins/CoinsSlice';
import coinsDetailsReducer from './details/CoinsDetails';

const store = configureStore({
  reducer: {
    coins: coinsReducer,
    coinsDetails: coinsDetailsReducer,
  },
});

export default store;
