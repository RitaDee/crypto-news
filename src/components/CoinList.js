import React from 'react';
import { useSelector } from 'react-redux';
import CoinUi from './CoinUi';

const CoinList = () => {
  const coins = useSelector((store) => store.coins.list);
  // const crypto = coins.name;
  // console.log(crypto);
  console.log(coins);

  return (
    <div className="coinDiv">
      <div className="conHeading">
        <h2>Crypto Market Prices</h2>
      </div>
      <div>
        {coins.map((coin) => (
          <CoinUi
            id={coins.coin.id}
            key={coins.coin.id}
            name={coins.coin.name}
            image={coins.coin.icon}
            currentPrice={coin.price}
          />
        ))}
      </div>
    </div>
  );
};

export default CoinList;
