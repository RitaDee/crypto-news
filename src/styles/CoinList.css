.coinsdiv {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  padding-block: 1rem;
  margin-inline: auto;
  height: 10vh;
  cursor: pointer;
  background: linear-gradient(90deg, rgba(100, 174, 186, 1) 0%, rgba(122, 172, 181, 1) 100%);
}

.coin-div {
  background-color: #fff;
  margin-top: -0.8rem;
  font-size: 10px;
  width: 100%;
}

.coinsheading {
  padding-left: 1rem;
  padding-top: 0.5rem;
  color: antiquewhite;
}
