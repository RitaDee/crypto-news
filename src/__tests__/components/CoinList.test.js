import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../Redux/store';
import CoinList from '../../components/CoinList';

describe('CoinList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CoinList />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
