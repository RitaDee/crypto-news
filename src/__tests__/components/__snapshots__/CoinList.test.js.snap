// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinList renders correctly 1`] = `
<div
  className="coin_container"
>
  <div
    className="coinsheading"
  >
    <h2
      className="h"
    >
      Crypto Market Prices
    </h2>
  </div>
  <div
    className="coinsdiv"
  />
</div>
`;
